### PR TITLE
add post comment endpoint

### PR DESCRIPTION
**Pull request by Miike-J**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -167,6 +167,58 @@ describe('/api/articles/:article_id/comments', () => {
...
+    test('404: article_id too high', () => {
+        const obj = {
+            username: 'rogersop',
+            body: 'fwifenffefm'
+        }
+        return request(app).post('/api/articles/140/comments').send(obj).expect(404).then(({body}) => {
+            expect(body.msg).toBe('bad request')
+        })
+    })
+    test('404: username not registered', () => {
```
Comment (k-huggs) [inline __tests__/app.test.js:213]: love the reference

File: __tests__/app.test.js [hunk 2]
```diff
@@ -167,6 +167,58 @@ describe('/api/articles/:article_id/comments', () => {
             expect(body.msg).toBe('article doesnt exist')
         })
     })
+    test('200: Post adds obj to comments table and returns posted obj', () => {
```
Comment (k-huggs) [inline __tests__/app.test.js]: double check your status codes for a post

File: Model/articles.model.js [hunk 3]
```diff
@@ -68,4 +68,14 @@ exports.selectArticleCommentsById = (article_id) => {
             return items
         })
     })
-}
\ No newline at end of file
+}
+
+exports.insertComment = (username, body, article_id, next) => {
```
Comment (k-huggs) [inline Model/articles.model.js]: why have you passsed next into your model function?

File: Controller/articles.contoller.js [hunk 4]
```diff
@@ -31,4 +31,13 @@ exports.getArticleCommentsById = (req, res, next) => {
...
     }).catch(next)
-}
\ No newline at end of file
+}
+
+exports.postArticleComment = (req, res, next) => {
+    const {username} = req.body
+    const {body} = req.body
+    const {article_id} = req.params
+    insertComment(username, body, article_id, next).then(results => {
```
Comment (k-huggs) [inline Controller/articles.contoller.js]: why have you passed next into the model?

Comment (Miike-J) [reply Controller/articles.contoller.js]: i thought i might need to do a .catch in the model but realised i didnt and forgot to remove it

File: __tests__/app.test.js [hunk 5]
```diff
@@ -167,6 +167,58 @@ describe('/api/articles/:article_id/comments', () => {
...
+                article_id: 1,
+                comment_id: expect.any(Number),
+                body: 'i dont know whats going on',
+                author: 'rogersop',
+                votes: 0,
+                created_at: expect.any(String)
+            }))
+        })
+    })
+    test('400: username/body are the wrong type', () => {
```
Comment (k-huggs) [inline __tests__/app.test.js:186]: could include missing fields 400 too? i.e. no username or body?

**Review by k-huggs — COMMENTED** [review]
Couple of changes to make!

**Pull request merged**